mercury-operator/src/main/java/com/redhat/mercury/operator/controller/ServiceDomainClusterController.java

```diff
@@ -154,7 +166,7 @@ private void createOrUpdateRole(ServiceDomainCluster sdc) {
             Role role = client.rbac().roles().createOrReplace(expected);
             LOGGER.debug("{} Role was missing, created. {}", SERVICE_DOMAIN_ROLE, role);
         } else {
-            if (!Objects.equals(current, expected)) {
+            if (!Objects.equals(current, expected) || !Objects.equals(current.getMetadata().getOwnerReferences(), expected.getMetadata().getOwnerReferences())) {
```

Comment:
Don't compare just current and expected because the status and metadata will be different. For the case of the role you only need to compare the rules, don't you think?

Reply:
Done